### PR TITLE
feat: 🎸 update new api and add all time rebalancing

### PR DESCRIPTION
**Pull request by hieunguyenebox**

### CODE REVIEW

**Comment by hieunguyenebox** [comment]
@tuan-tl @parrot9design  please review

**Review by parrot9design — CHANGES_REQUESTED** [review]
This is looking great! It doesn't seem like the data is being updated from the backend right now, as the values weren't changing.

I also noticed this glitch in the amount of padding when viewed in a smaller window: 

https://user-images.githubusercontent.com/63842577/126056750-dda58a96-aa9b-40c3-b6e6-05cb535e564d.mov

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@ generated-cra-app
 template
 .env
 .eslintcache
+# test
```
Comment (WeiLi512) [inline .gitignore:27]: What is it for?

File: src/store/loan/hooks.ts [hunk 2]
```diff
@@ -118,41 +120,29 @@ export function useLoanFetchInfo(account?: string | null) {
...
+      try {
+        const res = await queryRebalanced({
+          address: account,
+          symbol: 'bnUSD',
+          timestamp,
+        });
+
+        const { loan_repaid, collateral_sold } = res.data || {};
+        if (collateral_sold)
+          dispatch(
```
Comment (WeiLi512) [inline src/store/loan/hooks.ts:137]: why do you use redux?

Comment (WeiLi512) [reply src/store/loan/hooks.ts:137]: just use react-query. please reference the other part code.

File: src/app/components/home/PositionDetailPanel.tsx [hunk 3]
```diff
@@ -266,18 +283,22 @@ const PositionDetailPanel = () => {
...
-                  />
+                  <UnderlineTextWithArrow onClick={handleToggle} text={period} arrowRef={arrowRef} />
                   <DropdownPopper show={Boolean(anchor)} anchorEl={anchor} placement="bottom-end">
                     <MenuList>
                       {PERIODS.map(p => (
                         <MenuItem key={p} onClick={() => handlePeriod(p)}>
                           {p}
                         </MenuItem>
                       ))}
+                      <MenuItem
```
Comment (WeiLi512) [inline src/app/components/home/PositionDetailPanel.tsx:294]: Please try it.
`<MenuItem className="border-top" />`

**Review by WeiLi512 — CHANGES_REQUESTED** [review]

**Comment by hieunguyenebox** [comment]
@parrot9design , fixed, please check again

File: src/app/components/home/PositionDetailPanel.tsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import BigNumber from 'bignumber.js';
+import { timeStamp } from 'console';
```
Comment (WeiLi512) [inline src/app/components/home/PositionDetailPanel.tsx:4]: the var is useless.

File: src/queries/queryRebalancing.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import axios from 'axios';
```
Comment (WeiLi512) [inline src/queries/queryRebalancing.ts:1]: remove the file.

File: src/app/components/home/PositionDetailPanel.tsx [hunk 6]
```diff
@@ -62,19 +64,21 @@ const useOwnDailyRewards = (): BigNumber => {
...
-const PERIODS = [Period.day, Period.week, Period.month];
+const PERIODS = [Period.day, Period.week, Period.month, Period.all];
 
 const PositionDetailPanel = () => {
   const dailyRewards = useOwnDailyRewards();
   const rewardsAPY = useLoanAPY();
   const hasRewardableCollateral = useHasRewardableLoan();
   const upLarge = useMedia('(min-width: 1200px)');
   const [show, setShow] = React.useState<boolean>(false);
+  const [timestamp, setTimeStamp] = useState(dayjs().subtract(1, 'day').unix());
```
Comment (WeiLi512) [inline src/app/components/home/PositionDetailPanel.tsx:81]: calculate the `timestamp` from `period` var.

File: src/app/components/home/PositionDetailPanel.tsx [hunk 7]
```diff
@@ -126,7 +126,21 @@ const PositionDetailPanel = () => {
   const handlePeriod = (p: Period) => {
     closeMenu();
     setPeriod(p);
-    updateLoanTotalRepaid(p);
+
+    let timestamp = 0; // all
```
Comment (WeiLi512) [inline src/app/components/home/PositionDetailPanel.tsx:130]: put the code outside of this funtion.

**Review by WeiLi512 — APPROVED** [review]

File: src/app/components/home/PositionDetailPanel.tsx (hunk 6 above)
Comment (WeiLi512) [reply src/app/components/home/PositionDetailPanel.tsx:81]: .

**Review by WeiLi512 — CHANGES_REQUESTED** [review]

**Review by parrot9design — APPROVED** [review]

**Comment by BennyOptions** [comment]
This works for me now @tuan-tl. I checked and the numbers match

**Review by WeiLi512 — CHANGES_REQUESTED** [review]

**Review by WeiLi512 — APPROVED** [review]

**Pull request merged**